CAMBI-tech/BciPy
# Multi-modal acquisition

# Overview

Re-architected the data acquisition module in support of multi-modal acquisition.

## Ticket

https://www.pivotaltracker.com/story/show/178699590
https://www.pivotaltracker.com/story/show/178875719

## Contributions

- Created a new lightweight data acquisition client for reading from LSL data sources. Multiple clients can be created to read from different sources (ex. EEG and eye tracker).
- Separated out the responsibilities for recording and querying data.
- Deprecated MarkerWriter
- Deprecated support for TCP connections.
- Renamed `supported_device` to `preconfigured_device`.
- Added a Clock abstraction, which by default uses the pylsl local_clock.
- Refactored data queries to use timestamps rather than sample number.
- Added code to demonstrate functionality and usage.

## Test

- Run all unit tests
- Run Calibration task -> offline_analysis -> Copy Phrase task.
- Confirm timing data.
